python/cudf/cudf/dataframe/series.py

```diff
@@ -1795,6 +1801,52 @@ def round(self, decimals=0):
             index=self.index,
         )
 
+    def isin(self, test):
+
+        from cudf import DataFrame
+
+        lhs = self
+        rhs = None
+
+        try:
+            rhs = Series(columnops.as_column(test, dtype=self.dtype))
+        except Exception:
+            # pandas functionally returns all False when cleansing via
+            # typecasting fails
+            return Series(cudautils.zeros(len(self), dtype="bool"))
+
+        # If categorical, combine categories first
+        if pd.api.types.is_categorical_dtype(lhs):
+            lhs_cats = lhs.cat.categories.as_column()
+            rhs_cats = rhs.cat.categories.as_column()
+            if np.issubdtype(rhs_cats.dtype, lhs_cats.dtype):
+                # if the categories are the same dtype, we can combine them
+                cats = Series(lhs_cats.append(rhs_cats)).drop_duplicates()
+                lhs = lhs.cat.set_categories(cats, is_unique=True).fillna(-1)
+                rhs = rhs.cat.set_categories(cats, is_unique=True).fillna(-1)
```

Comment:
Do we set to -1 just to match Pandas behavior? Should we leave as null instead for matching Arrow?

Reply:
I modeled this logic after the the way `join` merges the categories. I think the fillnas can be removed.